xclim/utils.py

```diff
@@ -975,3 +975,108 @@ def wrapped_partial(func, *args, **kwargs):
     partial_func = partial(func, *args, **kwargs)
     update_wrapper(partial_func, func)
     return partial_func
+
+
+def uas_vas_2_wind(uas, vas):
+    """Converts uas and vas to the wind speed sfcWind
+
+    Parameters
+    ----------
+    uas : xr.DataArray
+      eastward wind (m s-1)
+    vas : xr.DataArray
+      northward wind (m s-1)
+
+    Returns
+    -------
+    wind : xr.DataArray
+      wind speed (m s-1)
+    winddir : xr.DataArray
+      direction FROM WHICH the wind blows (corresponds to the meteorological standard)
+    """
+    # TODO: This needs to be tested with 2D data.
+
+    # Wind speed is the Pythagore of "uas" and "vas"
+    wind = np.sqrt(np.square(uas) + np.square(vas))
+
+    # Add attributes to wind. This is done by copying uas' attributes and overwriting a few of them
+    wind.attrs = uas.attrs
+    wind.name = 'wind'
+    wind.attrs['standard_name'] = 'wind_speed'
+    wind.attrs['long_name'] = 'Wind Speed'
+
+    # Calculate the angle
+    # TODO: This creates decimal numbers such as 89.99992. Do we want to round?
+    winddir_math = np.degrees(np.arctan2(vas, uas))
+
+    # Convert the angle from the mathematical standard to the meteorological standard
+    winddir = (270 - winddir_math) % 360.0
+
+    # According to the meteorological standard, calm winds must have a direction of 0°, while northerly winds have a direction of 360°.
+    # On the Beaufort scale, calm winds are defined as < 0.5 m/s
+    # TODO: units check for wind speed in m/s
+    # TODO: ideally, we would want to circumvent ".values", as this forces xarray to do calculations
+    if wind.attrs['units'] in ['m/s', 'm s-1']:
+        winddir.values[(winddir.round() == 0) & (wind >= 0.5)] = 360.0
+        winddir.values[wind < 0.5] = 0
+    else:
+        raise Exception('Units need to be in m/s!')
+
+    # Add attributes to winddir. This is done by copying uas' attributes and overwriting a few of them
+    winddir.attrs = uas.attrs
+    winddir.name = 'winddir'
+    winddir.attrs['standard_name'] = 'wind_from_direction'
+    winddir.attrs['long_name'] = 'Direction from which the Wind Blows'
+    winddir.attrs['units'] = 'degree'
+
+    return wind, winddir
+
+
+def wind_2_uas_vas(wind, winddir):
+    """Converts sfcWind to the components uas and vas
+
+    Parameters
+    ----------
+    wind : xr.DataArray
+      wind speed (units do not need to be in CF-Convention)
+    winddir : xr.DataArray
+      direction FROM WHICH the wind blows (corresponds to the meteorological standard)
```

Comment:
Direction from which the wind blows, following the meteorological convention where 360 stands for North.